plugins/score-card/src/api/ScoringDataApi.ts

```diff
@@ -22,6 +22,13 @@ export const scoringDataApiRef = createApiRef<ScoringDataApi>({
 });
 
 export type ScoringDataApi = {
-  getScore(entity?: Entity): Promise<EntityScoreExtended | undefined>;
-  getAllScores(entityKindFilter?: string[]): Promise<EntityScoreExtended[] | undefined>;
+  getScore(
+    entity?: Entity,
+    auth?: any,
```

Comment:
Not sure about this. What kind of authentication it is? I mean for your purposes GitHub may fits well, others would invite Azure DevOps authentication... the actual provider depends on the url you want to integrate/download data from.

Reply:
I would recommend do it via dependency injection. Like finding a suitable provider or at least specify own provider/interface you need to consume like

```typeScript
# what you want to inject...SCM stands for Source Control Manager
interface IScmProvider {
  getRawFile(project: string, relativePath: string): Promise<string>;
}
# ScoringApi would look like
export type ScoringDataApi = {
  getScore(
    entity?: Entity,
    scmProvider?: IScmProvider,
  ): Promise<EntityScoreExtended | undefined>;
```

Then anyone would be able to provide their own implementation IScmProvider. And as a default you may provide the GH implementation.

It also should nicely separate the loading/handlign logic and the underlying SCM.
Maybe there is already some existing FE provider, but so far I have found only backend providers (meaning we would need to provide the backend part for the scoring plugin) e.g. [EntityProvider](https://github.com/Oriflame/backstage/blob/4285d5824f13b0305684bb4cb91e65c03aded59d/plugins/catalog-backend-module-github/src/providers/GithubEntityProvider.ts#L74)

Reply:
Maybe [ScmIntegrationsApi](https://github.com/Oriflame/backstage/blob/4285d5824f13b0305684bb4cb91e65c03aded59d/packages/integration-react/src/api/index.ts#L25) could be used.

There is also `discoveryApi`, `fetchApi`, `identityApi`...

Also thing of, every entity might be theoretically saved to a different location thus using a different SCM. Can we somehow distinguish from the `Location` of the `Entity`?

Reply:
I am finding it hard to make it generic that works for all SCMs. Backstage does not provide `URLReaders` like component for FE.  Can we just tweak this PR to support just GH? That is if repo contains the mentioned annotations it would use `gitHubApiRef` else will fall back to default implementation.

Reply:
This improvement is in place now. It has a urlReader like functionality that could be extended to other sources if needed.